core/ceremony/ceremony.go

```diff
@@ -264,21 +296,33 @@ func (vc *ValidationCeremony) handleFlipLotteryPeriod(block *types.Block) {
 }
 
 func (vc *ValidationCeremony) handleShortSessionPeriod(block *types.Block) {
-	timestamp := vc.epochDb.ReadShortSessionTime()
-	if timestamp != nil {
-		vc.appState.EvidenceMap.SetShortSessionTime(timestamp, vc.config.Validation.GetShortSessionDuration())
-	} else if block.Header.Flags().HasFlag(types.ShortSessionStarted) {
-		t := time.Now().UTC()
-		vc.epochDb.WriteShortSessionTime(t)
-		vc.appState.EvidenceMap.SetShortSessionTime(&t, vc.config.Validation.GetShortSessionDuration())
-		if vc.shouldInteractWithNetwork() {
-			vc.logInfoWithInteraction("Short session started", "at", t.String())
-		}
+	if block.Header.Flags().HasFlag(types.ShortSessionStarted) {
+		vc.startShortSession()
 	}
 	vc.broadcastFlipKey()
 	vc.processCeremonyTxs(block)
 }
 
+func (vc *ValidationCeremony) startShortSession() {
+	if vc.shortSessionStarted {
+		return
+	}
+	vc.validationStartMutex.Lock()
+	defer vc.validationStartMutex.Unlock()
+
+	if vc.appState.State.ValidationPeriod() < state.FlipLotteryPeriod {
+		return
+	}
+
+	t := vc.appState.State.NextValidationTime()
+	vc.appState.EvidenceMap.SetShortSessionTime(&t, vc.config.Validation.GetShortSessionDuration())
```

Comment:
maybe we can do it when evidence map is initialized\reinitialized, params here are static.